docs/wiki/Vision.md

```diff
@@ -0,0 +1,115 @@
+1. The resource group
+1. The virtual network & storage account
+1. The virtual Machine
+
+Then we'd only need to create a parameter file for the orchestration-template and have the workflow deploy both in combination.
+
+### What is the intended deployment flow?
+
+In this section we'll take a deeper look into the fundamental flow of the platform.
+
+First things first, we would work towards the deployment of our environments in 3 phases:
+
+<img src="media/deploymentFlow.png" alt="Deployment flow" height="150">
+
+The **Produce components** phase is the phase where you're supposed to set up your modules and validate that they work. In case of _CARML_ this means we have a pipeline for each module that, with one or multiple test-parameter files, runs static as well as validation & deployment tests on its template. Once if all tests pass we can be certain no bugs were introduced (in case the template was altered) and all features work as intended.
+
+The next phase, **Publish artifacts**, will take the tested and approved modules and publish them to a target location of your choice (for example _template specs_ or the _bicep registry_). The publishing should publish at least the tested module template itself.
+The target location should support versioning so that you only always publish new versions.
+This may bring up the question _Why_ you should publish your templates before you use them. Afterall, you could just reference the modules in the source repository as is to deploy you environments. This however has one major drawback: If your deployments rely on what you have in your source repository then they 'by definition' always use the latest code. This means, as soon as somebody introduces a breaking change to one of these templates, all your orchestrated and referencing deployments will be blocked until you updated them to the latest version too.
+By introducing versions to you modules, the referencing templates can and should specify a specific module versions they want to use and deploy their environment with those. If we now have the case that a breaking change is introduced, a new version is published, but no deployment is affected because they still reference the original. Instead they have to make the deliberate decision to upgrade the module references to newer versions.
+
+The last phase then is the **artifact consumption**. As mentioned previously, this is the phase where we orchestrate the deployment of for example services, workloads or entire landing zones. Up to this point, all deployments were only done for test reasons and should be deleted after a successful run. The deployments we perform now are instead 'sticky' deployments to for example a integration or production environment. By referencing and orchestrating the templates published in the previous step we can be sure that all templates work and we only need to provide them with the correctly configured parameters and orchestrate their deployment in the correct order.
```

Comment:
Consider rephrasing to simplify the point you're making

Reply:
done